tests/test_serialization.py

```diff
@@ -627,3 +642,161 @@ class RootContainer:
+        assert dumped == [{"value": 123}, {"value": 456}]
+
+
+def test_generic_in_parents() -> None:
+    _TXxx = TypeVar("_TXxx")
+    _TData = TypeVar("_TData")
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class Data(Generic[_TXxx]):
+        xxx: _TXxx
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class ParentClass(Generic[_TData]):
+        value: str
+        data: _TData
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class ChildClass(ParentClass[Data[int]]):
+        pass
+
+    instance = ChildClass(value="vvv", data=Data(xxx=111))
+    dumped = mr.dump(instance)
+
+    assert dumped == {"value": "vvv", "data": {"xxx": 111}}
+    assert mr.load(ChildClass, dumped) == instance
+
+
+def test_generic_type_var_with_reuse() -> None:
+    _T = TypeVar("_T")
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class T1(Generic[_T]):
+        t1: _T
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class T2(Generic[_T], T1[int]):
+        t2: _T
+
+    instance = T2[str](t1=1, t2="2")
+
+    dumped = mr.dump(instance, cls=T2[str])
+
+    assert dumped == {"t1": 1, "t2": "2"}
+    assert mr.load(T2[str], dumped) == instance
+
+
+def test_generic_with_field_override() -> None:
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class Value1:
+        v1: str
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class Value2(Value1):
+        v2: str
+
+    _TValue = TypeVar("_TValue", bound=Value1)
+    _TItem = TypeVar("_TItem")
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class T1(Generic[_TItem]):
+        value: Value1
+        iterable: Iterable[_TItem]
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class T2(Generic[_TValue, _TItem], T1[_TItem]):
+        value: _TValue
+        iterable: set[_TItem]
+
+    instance = T2[Value2, int](value=Value2(v1="aaa", v2="bbb"), iterable=set([3, 4, 5]))
+
+    dumped = mr.dump(instance, cls=T2[Value2, int])
+
+    assert dumped == {"value": {"v1": "aaa", "v2": "bbb"}, "iterable": [3, 4, 5]}
+    assert mr.load(T2[Value2, int], dumped) == instance
+
+
+def test_generic_reuse_with_different_args() -> None:
+    _TItem = TypeVar("_TItem")
+
+    @dataclasses.dataclass(frozen=True, slots=True, kw_only=True)
+    class GenericContainer(Generic[_TItem]):
+        items: list[_TItem]
+
+    container_int = GenericContainer[int](items=[1, 2, 3])
+    dumped = mr.dump(container_int, cls=GenericContainer[int])
+
+    assert dumped == {"items": [1, 2, 3]}
+    assert mr.load(GenericContainer[int], dumped) == container_int
+
+    container_str = GenericContainer[str](items=["q", "w", "e"])
+    dumped = mr.dump(container_str, cls=GenericContainer[str])
+
+    assert dumped == {"items": ["q", "w", "e"]}
+    assert mr.load(GenericContainer[str], dumped) == container_str
+
+
+def test_sdfdfsd():
```

Comment:
![image](https://github.com/user-attachments/assets/e0794b92-a03c-42ad-9edb-ddff788f8aa2)

Reply:
removed.

I tested examples for readme.md https://github.com/anna-money/marshmallow-recipe/pull/163/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R81

but forgot to remove them